src/components/MessagesList/MessagesList.vue

```diff
@@ -29,26 +29,34 @@ get the messagesList array and loop through the list to generate the messages.
 	<!-- size and remain refer to the amount and initial height of the items that
 	are outside of the viewport -->
 	<div ref="scroller"
+		:key="token"
 		class="scroller messages-list__scroller"
-		:class="{'scroller--chatScrolledToBottom': isChatScrolledToBottom}"
-		@scroll="debounceHandleScroll">
+		:class="{'scroller--chatScrolledToBottom': isChatScrolledToBottom,
+			'scroller--isScrolling': isScrolling}"
+		@scroll="onScroll"
+		@scrollend="endScroll">
 		<div v-if="displayMessagesLoader" class="scroller__loading icon-loading" />
 
-		<template v-for="item of messagesGroupedByAuthor">
-			<div v-if="item.dateSeparator" :key="`date_${item.id}`" class="messages-group__date">
+		<ul v-for="(list, dateTimestamp) in messagesGroupedByDateByAuthor"
+			:key="`section_${dateTimestamp}`"
+			:date-timestamp="dateTimestamp"
+			class="dates-group-wrapper"
+			:class="{'has-sticky': dateTimestamp === stickyDate}">
+			<li class="messages-group__date">
 				<span class="messages-group__date-text" role="heading" aria-level="3">
-					{{ item.dateSeparator }}
+					{{ generateDateSeparator(dateTimestamp) }}
```

Comment:
Or not. If you keep the browser overnight, it won't change from Today to Yesterday..

Reply:
it was sthg like that before, I will revert to it.

Reply:
UPD: will be updated after any messagesList change or chat rejoining next day

Reply:
But if you are right in the previous comment : If you keep same chat open and not new message or update came in, Today will remain instead of Yesterday.

Reply:
When need to investigate, why it's triggered by scroll

Reply:
I think it's because of Vue re-rendering behavior when key is present. We can make a small tweak: set currentDay prop and watch over it. If it changes, then we regenerate the array of dataSeps.